tests/check_warnings.py

```diff
@@ -0,0 +1,56 @@
+from pathlib import Path
+import sys
+
+from colorama import Fore, init
+
+# init colors for all plateforms
+init()
+
+
+def check_warnings(file):
+    """
+    Check the list of warnings produced by the GitHub CI tests
+    raise errors if there are unexpected ones and/or if some are missing
+
+    Args:
+        file (pathlib.Path): the path to the generated warning.txt file from
+            the CI build
+
+    Return:
+        0 if the warnings are all there
+        1 if some warning are not registered or unexpected
+    """
+
+    # print some log
+    print("\n=== Sphinx Warnings test ===\n\n")
+
+    # find the file where all the known warnings are stored
+    warning_file = Path(__file__).parent / "warning_list.txt"
+
+    test_warnings = file.read_text().strip().split("\n")
+    ref_warnings = warning_file.read_text().strip().split("\n")
+
+    # find all the missing warnings
+    missing_warnings = []
+    for w in ref_warnings:
+        index = [i for i, wi in enumerate(test_warnings) if w in wi]
```

Comment:
non-blocking comment, but just a random tip: I've found it useful to use double letters instead of single letters for iterations (so, use `ii` instead of `i`, or use `wa` instead of `w` for "warn"). This is because it is much easier to `ctrl+f` for `ii` and find something meaningful, than to do so with just `i`.

Reply:
nice trick, I'll keep that for other projects !